bionic/deriver.py

```diff
@@ -181,47 +184,48 @@ def _initialize_task_state(self, task_state):
 
         # All names in this task should point to the same provider.
         task_state.provider, = set(
-            self._flow_state.get_provider(task_key.entity_name)
+            self._flow_state.get_provider(task_key.dnode.to_entity_name())
```

Comment:
This is related to the following comment but I wanted to expand a bit here: I've left `FlowState.get_provider` taking an entity name, not a generic descriptor, because a flow state is a mapping from entities (not descriptors) to providers. In the future the interface will probably change, but I'm not sure how yet.